src/pybind/mgr/dashboard/controllers/cephfs.py

```diff
@@ -722,10 +722,13 @@ def set(self, vol_name: str, subvol_name: str, size: str):
 
         return f'Subvolume {subvol_name} updated successfully'
 
-    def delete(self, vol_name: str, subvol_name: str):
+    def delete(self, vol_name: str, subvol_name: str, retain_snapshots: bool = False):
+        params = {'vol_name': vol_name, 'sub_name': subvol_name}
+        retain_snapshots = str_to_bool(retain_snapshots)
+        if retain_snapshots:
+            params['retain_snapshots'] = 'True'
         error_code, _, err = mgr.remote(
-            'volumes', '_cmd_fs_subvolume_rm', None, {
-                'vol_name': vol_name, 'sub_name': subvol_name})
+            'volumes', '_cmd_fs_subvolume_rm', None, params)
         if error_code != 0:
             raise RuntimeError(
```

Comment:
this should be a `DashboardException` so that the error will be properly handled by dashboard. Otherwise we would only get a 500 error.
```suggestion
            raise DashboardException(
```